pkg/cleaner/cleaning.go

```diff
@@ -0,0 +1,138 @@
+package cleaner
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/must-gather-clean/pkg/kube"
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/omitter"
+)
+
+type CleaningProcessor interface {
+	// ProcessFile is the end2end method for cleaning (omit + obfuscate path and content of) a file on disk.
+	// Will return nil if the file was processed without an error (e.g. through omission) or the error otherwise.
+	ProcessFile(inputFile string) error
+}
+
+type ReaderWriterObfuscator interface {
+	// ObfuscateInputReader obfuscates on an agnostic line-based reader and writes to an agnostic writer facility
+	ObfuscateInputReader(inputReader io.Reader, outputWriter io.Writer) error
+}
+
+type FileObfuscator interface {
+	// ObfuscatePlainTextFile obfuscates a text file and writes the result into the outputFile.
+	ObfuscatePlainTextFile(inputFile string, outputFile string) error
+}
+
+type ContentObfuscator struct {
+	Obfuscator obfuscator.ReportingObfuscator
+}
+
+type FileCleaner struct {
+	ContentObfuscator
+
+	inputFolder  string
+	outputFolder string
+
+	omitter omitter.ReportingOmitter
```

Comment:
The omitter tracks what's omitted. But the FileProcessor actually omits the file. This seems wrong to me.

Reply:
it was how the logic was before and not the goal of the refactoring. 
AFAIR you've added that interface with that behaviour :)

Reply:
No, the `Omiter` didn't have any reporting functionality for the reason mentioned above.